_posts/2019/2019-10-10-building-a-drawer-component-in-react.md

```diff
@@ -0,0 +1,805 @@
+
+Here's the CSS we'll need to achieve the effect:
+
+``` scss
+.Drawer {
+  $arrowSize: 5px;
+
+  &-arrow {
+    border-top: $arrowSize solid transparent;
+    border-bottom: $arrowSize solid transparent;
+    border-left: $arrowSize solid #454545;
+    width: 0;
+    height: 0;
+  }
+}
+```
+
+A quick explanation of the above:
+
+* We set the element to have `height` and `width` of 0. This is because adding a border
+to a 0 sized element makes the borders meet at the center of the element like a triangle.
+* `border-top` and `border-bottom` are set to transparent to hide the top and bottom of the triangle
+* `border-left` adds a border on the left side but because it is triangular it appears to be
+pointing to the right.
+
+![Drawer arrow broken](/img/2019/drawer-arrow-off.png)
+
+We've now got a triangle indicator but it isn't positioned properly as seen above. Let's move this to
+the right side of the button using Flexbox.
+
+* First we'll give the parent element `.Drawer-title` the property of `display: flex`.
+* Then we'll set the actual title text element above to flex as much as possible using `flex: 1`.
+* Finally, a small tweak to our arrow by using `align-self: center` make it appear on the same baseline as the bottom of our text characters. Here's the resulting CSS:
+
+
+``` scss
+.Drawer {
+
+  &-title {
+    // omitted
+
+    display: flex;
+  }
+
+  &-titleText {
+    flex: 1;
+  }
+
+  &-arrow {
+    border-top: $arrowSize solid transparent;
+    border-bottom: $arrowSize solid transparent;
+    border-left: $arrowSize solid #454545;
+    width: 0;
+    height: 0;
+
+    align-self: center;
+  }
+}
+```
+
+We can also go one more step and animate this alongside the content animation. This
+will give the appearance of the triangle twisting downward making it look like the drawer
+is expanding.
+
+``` scss
+.Drawer {
+
+  &-arrow {
+    border-top: $arrowSize solid transparent;
+    border-bottom: $arrowSize solid transparent;
+    border-left: $arrowSize solid #454545;
+    width: 0;
+    height: 0;
+
+    align-self: center;
+    transition: all .5 ease;
+  }
+
+  &.is-expanded {
+    .Drawer-arrow {
+      transform: rotate(90deg);
+    }
+  }
+}
+```
+
+And here's our final iteration for this article:
+
+![Final Drawer Component](/img/2019/final-drawer.gif)
+
+## Wrapping up
+
+So running through what we accomplished above here's what we've produced:
+
+* We created a new component that has a onClick event to set a CSS class
+* We're calculating the hidden content's height in JavaScript
+* Using CSS we've hidden our content drawer, animated the height change, and created
+a triangle.
+
+If you'd like to see a full versions of the code above, [I've condensed it here](/doc/2019/drawer-final.txt).
+
+I'll be writing a follow-up article to this one explaining how we can convert the above
+into a Compound Component for better compasibility. As always, thanks for reading.
+
+*[DRY]: Don't Repeat Yourself
```

Comment:
couple of ideas. 
1 give the drawer some creative items/titles ie call it a clothing drawer and make your items socks, shirts, sweaters. then you talk about taking or adding things to drawer. this would be especially helpful for a baby dev and it make it more interesting to read. 
2. i would read through this blog and take yourself back to beginning a jr dev. are there some areas you could add some links or extra info to help those new the craft? you're starting to write like a sr. dev which isn't a bad thing but might limit your readership. 
3. watch for introduction clauses. you like to use two as place ex. the first thing we'll do is start by..... instead of just saying we'll start with....